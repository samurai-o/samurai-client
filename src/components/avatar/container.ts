import styled from 'styled-components';

export const AvatarContainer = styled.div`
  border-radius: 10px;
  cursor: pointer;
  overflow: hidden;
`;
