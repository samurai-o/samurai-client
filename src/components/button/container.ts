import styled from 'styled-components';
import { ButtonType, colors } from '.';

export type ButtonStyledContainer = {
  color: ButtonType;
};

export const ButtonContainer = styled.button<ButtonStyledContainer>`
  padding: 4px 10px;
  border-radius: 4px;
  border: transparent;
  outline: none;
  cursor: pointer;
  color: ${({ color }) => (color !== 'text' ? '#fff' : '#000')};
  background: ${({ color }) => colors[color]};
`;
