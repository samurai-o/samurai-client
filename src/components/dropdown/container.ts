import styled from 'styled-components';
import { Placements } from '.';

export const DropdownContainer = styled.div`
  position: relative;
`;

export type DropdownBoxStyleProps = {
  show?: boolean;
  placements: Placements;
};
export const DropdownBoxContainer = styled.div<DropdownBoxStyleProps>`
  min-width: 100px;
  position: absolute;
  top: calc(100% + 2px);
  padding-top: 4px;
  padding-bottom: 4px;
  background: #fff;
  left: ${({ placements }) =>
    placements === 'left'
      ? '0px'
      : placements === 'center'
      ? 'calc(50% - 4px)'
      : 'unset'};
  right: ${({ placements }) => (placements === 'right' ? '0px' : 'unset')};
  transition: opacity 250ms ease, transform 250ms ease;
  box-shadow: 0px 4px 8px rgb(0 0 0 / 10%);
  backface-visibility: hidden;
  border-radius: 4px;
  border: 1px solid #d0d7de;
  pointer-events: ${({ show }) => (!!show ? 'all' : 'none')};
  opacity: ${({ show }) => Number(!!show)};
  transform: ${({ show }) =>
    !!show
      ? 'translate3d(0, 0, 0)'
      : 'translate3d(0, 32px, 0) rotateX(-16deg) rotateY(14deg)'};
  &::after {
    position: absolute;
    bottom: 100%;
    left: ${({ placements }) =>
      placements === 'left'
        ? '8px'
        : placements === 'center'
        ? 'calc(50% - 4px)'
        : 'unset'};
    right: ${({ placements }) => (placements === 'right' ? '8px' : 'unset')};
    content: '';
    transform: translateY(4px) rotate(45deg);
    width: 8px;
    height: 8px;
    border: 1px solid #d0d7de;
    border-right: none;
    border-bottom: none;
    background-color: #fff;
  }
`;
