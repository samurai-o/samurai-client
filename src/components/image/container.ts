import styled from 'styled-components';

export const ImageContainer = styled.div<any>`
  width: ${({ width }) => (typeof width === 'number' ? `${width}px` : '100px')};
  border-radius: 8px;
  overflow: hidden;
`;

export const ImageDom = styled.img`
  width: inherit;
  vertical-align: middle;
`;
