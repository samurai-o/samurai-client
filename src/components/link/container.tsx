import styled from 'styled-components';

export const LinkButton = styled.a`
  width: fit-content;
  text-decoration: none;
  display: block;
  cursor: pointer;
`;
