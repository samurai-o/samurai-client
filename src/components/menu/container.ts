import styled from 'styled-components';
import { Overall } from '.';

export const MenuContainer = styled.div``;

export type MenuListStyleProps = {
  noEvent?: boolean;
  overall: Overall;
};

export const MenuList = styled.ul<MenuListStyleProps>`
  list-style: none;
  margin: unset;
  padding: unset;
  display: flex;
  flex-direction: ${({ overall }) =>
    overall === 'vertical' ? 'column' : 'row'};
`;

export type MenuItemStyleProps = {
  noEvent?: boolean;
  tag?: string | number;
};

export const MenuItem = styled.li<MenuItemStyleProps>`
  list-style: none;
  margin: unset;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 4px;
  overflow: hidden;
  margin: 0 4px;
  transition: all 250ms ease;
  &:hover {
    background: #e0dede47;
  }
  &:active {
    background: #ddd;
  }
`;
