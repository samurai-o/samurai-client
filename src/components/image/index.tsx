import React from 'react';
import { Size } from '../../common/interfaces';
import { ImageContainer, ImageDom } from './container';

export type ImageSize = Size;

export type ImageProps = {
  width?: number;
  type?: ImageSize;
  src?: string;
};

/**
 * 图片
 * @param props
 * @returns
 */
export function Image(props: ImageProps) {
  const { src, ..._props } = props;
  if (typeof src !== 'string') return null;
  return (
    <ImageContainer {..._props}>
      <ImageDom src={src} />
    </ImageContainer>
  );
}
