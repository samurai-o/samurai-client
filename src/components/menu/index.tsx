import React, { useMemo } from 'react';
import { isArray, isEmpty, isFunc } from '@frade-sam/samtools';
import { MenuContainer, MenuItem, MenuList } from './container';

// 布局方式
export type Overall = 'horizontal' | 'vertical';

export type MenuProps = {
  overall?: Overall;
  onClick?: (key: string) => void;
  children?: JSX.Element | JSX.Element[];
};

/**
 * 菜单数据配置
 * @returns
 */
export function Menu(props: MenuProps) {
  const { overall = 'vertical', children = [], onClick } = props;

  // 选项节点
  const MenuNodes = useMemo(() => {
    if (isEmpty(children)) return [];
    const childs: JSX.Element[] = !Array.isArray(children)
      ? [children]
      : children;
    return (
      <MenuList overall={overall}>
        {React.Children.map(childs, (child) => {
          if (child.type !== MenuItem) return null;
          const { onClick: onChildClick, tag, ..._props } = child.props;
          return React.cloneElement(child, {
            ..._props,
            onClick: () => {
              if (isFunc(onChildClick)) onChildClick(tag);
              if (typeof onClick === 'function') onClick(tag);
            },
          });
        })}
      </MenuList>
    );
  }, [children, overall]);

  return <MenuContainer>{MenuNodes}</MenuContainer>;
}

Menu.Item = MenuItem;
