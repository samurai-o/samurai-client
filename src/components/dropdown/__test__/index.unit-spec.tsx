import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Dropdown } from '..';
import { Image } from '../../image';
import { Menu } from '../../menu';
import { AdminBtn } from '../../../common/modules/admin';
import { Text } from '../../text';
import { DropdownBoxContainer, DropdownContainer } from '../container';

describe('avatar 组件', () => {
  it('快照测试', () => {
    const node = create(
      <Dropdown
        trigger="hover"
        menus={
          <Menu>
            <Menu.Item>文档</Menu.Item>
            <Menu.Item>文档2</Menu.Item>
          </Menu>
        }
      >
        <div>1</div>
      </Dropdown>,
    );
    expect(node).toMatchSnapshot();
  });

  it('trigger', () => {
    const node = shallow(
      <Dropdown
        trigger="hover"
        menus={
          <Menu overall="vertical">
            <Menu.Item tag="docs">
              <Text type="text">文档</Text>
            </Menu.Item>
            <Menu.Item tag="num">
              <Text type="text">文档</Text>
            </Menu.Item>
          </Menu>
        }
      >
        <AdminBtn />
      </Dropdown>,
    );
    expect(node.children().length).toEqual(2);
    node.find(AdminBtn).simulate('click');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(false);
    node.find(AdminBtn).simulate('mouseover');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(true);
    node.find(AdminBtn).simulate('mouseleave');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(false);
  });

  it('empty children', () => {
    const node = shallow(
      <Dropdown
        trigger="hover"
        menus={
          <Menu overall="vertical">
            <Menu.Item tag="docs">
              <Text type="text">文档</Text>
            </Menu.Item>
            <Menu.Item tag="num">
              <Text type="text">文档</Text>
            </Menu.Item>
          </Menu>
        }
      />,
    );
    expect(node.children().length).toEqual(1);
  });

  it('empty menus', () => {
    const node = shallow(
      <Dropdown trigger="hover">
        <AdminBtn />
      </Dropdown>,
    );
    expect(node.find(DropdownBoxContainer).children().isEmptyRender()).toEqual(
      true,
    );
  });

  it('children click', () => {
    const node = shallow(
      <Dropdown trigger="click">
        <AdminBtn />
      </Dropdown>,
    );
    expect(
      !!node.find(DropdownContainer).children().first().prop('onClick'),
    ).toEqual(true);
    node.find(DropdownContainer).children().first().simulate('click');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(true);
  });

  it('children mouse', () => {
    const node = shallow(
      <Dropdown trigger="hover">
        <AdminBtn />
      </Dropdown>,
    );
    node.find(DropdownContainer).children().first().simulate('mouseover');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(true);
    node.find(DropdownContainer).children().first().simulate('mouseleave');
    expect(node.find(DropdownBoxContainer).prop('show')).toEqual(false);
  });
});
