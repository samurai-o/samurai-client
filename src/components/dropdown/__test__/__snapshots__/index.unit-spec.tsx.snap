// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar 组件 快照测试 1`] = `
<div
  className="sc-bdvvaa fKRyFQ"
>
  <div
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    1
  </div>
  <div
    className="sc-gsDJrp xYbNh"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="sc-dkPtyc liIQBr"
    >
      <ul
        className="sc-hKwCoD eAINRV"
      >
        <li
          className="sc-eCImvq dontXv"
          onClick={[Function]}
        >
          文档
        </li>
        <li
          className="sc-eCImvq dontXv"
          onClick={[Function]}
        >
          文档2
        </li>
      </ul>
    </div>
  </div>
</div>
`;
